### ISSUE TITLE
HAProxy defaults provided from within Rancher UI don't have the correct indentation and won't override existing settings

### ISSUE THREAD

**Issue by geertvanheusden** [rancher/rancher#4403]
Rancher Version:
**1.0.0**

Docker Version:
**1.10.0**

OS: 
**Debian**

Steps to Reproduce: 
Update the LB service with some defaults
![image](https://cloud.githubusercontent.com/assets/1923423/14519428/1810b55a-021f-11e6-9c43-7c9f7e7c6b18.png)

Results: 
Check `/etc/haproxy/haproxy.cfg` in the LB container:

```
defaults
        log     global
        mode    tcp
        option  tcplog
        option  dontlognull
        option  redispatch
        option http-server-close
        option forwardfor
        retries 3
        timeout connect 5000
        timeout client 50000
        timeout server 50000
        errorfile 400 /etc/haproxy/errors/400.http
        errorfile 403 /etc/haproxy/errors/403.http
        errorfile 408 /etc/haproxy/errors/408.http
        errorfile 500 /etc/haproxy/errors/500.http
        errorfile 502 /etc/haproxy/errors/502.http
        errorfile 503 /etc/haproxy/errors/503.http
        errorfile 504 /etc/haproxy/errors/504.http
        timeout client 2m
timeout server 2m
```

Expected:
- Correct indentation of the provided second default setting
- The `timeout server` and `timeout client` settings are set twice in the defaults section

**Comment by alena1108** [comment]
@geertvanheusden the last setting overrides the first on haproxy, so its ok to see the duplicates. 

For the first round of haproxy config modification, we've decided to keep things simple and just append settings defined on rancher, to the backend, and that setting will just overrides the defaults. With the planned LB refactoring - where we plan to introduce overriding even listeners and backend section - we are going to rework the way we merge default/user defined settings, so the duplicates will be gone, and the haproxy config will look cleaner.

Closing as a dup of https://github.com/rancher/rancher/issues/2179
